app/routes/devices.$ip_addr.tsx

```diff
@@ -0,0 +1,143 @@
+import {
+  Outlet,
+  Link,
+  useParams,
+  useLoaderData,
+  useLocation,
+} from "@remix-run/react";
+import { LoaderFunctionArgs, json } from "@remix-run/node";
+import { createClient } from "@hey-api/client-fetch";
+import * as Evolver from "client/services.gen";
+import clsx from "clsx";
+import { EvolverConfigWithoutDefaults } from "client";
+import { BeakerIcon } from "@heroicons/react/24/outline";
+import { ENV } from "~/utils/env.server";
+
+export const handle = {
+  breadcrumb: ({ params }: { params: { ip_addr: string } }) => {
+    const { ip_addr } = params;
+    return <Link to={`/devices/${ip_addr}/state`}>{ip_addr}</Link>;
+  },
+};
+
+export async function loader({ params }: LoaderFunctionArgs) {
+  const { ip_addr } = params;
+  const evolverClient = createClient({
+    baseUrl: `http://${ip_addr}:${ENV.DEFAULT_DEVICE_PORT}`,
```

Comment:
I read this as the device identifier is `ip_addr`, but wondering if this might be rather `{ip_addr}:{port}`, since the port served by another device could be different from the default.

Reply:
yeah that's a good point, or a hash of ip, port and the name... Ultimately it's the name of the evolver not really its address that matters. Later evolvers could be at urls too. I'll have a think about refactoring this.

Reply:
cool - can do in a separate issue, of course